test/parallel/test-zlib-from-gzip-with-trailing-garbage.js

```diff
@@ -28,9 +32,12 @@ data = Buffer.concat([
   Buffer(10).fill(0)
 ]);
 
-assert.throws(() => zlib.gunzipSync(data));
+assert.throws(
+  () => zlib.gunzipSync(data),
+  /^Error: unknown compression method$/
+);
 
-zlib.gunzip(data, common.mustCall((err, result) => {
+zlib.gunzip(data, common.mustCall((err) => {
   assert(err);
```

Comment:
I would also recommend restoring `result` as an argument and adding an assertion for it. So all together, maybe something like this?:

```js
assert(err instanceof Error);
assert.strictEqual(err.code, 'Z_DATA_ERROR');
assert.strictEqual(err.message, 'unknown compression method');
assert.strictEqual(result, undefined);
```